### PR TITLE
Migrator - CSC specific migration of v1 records

### PR DESCRIPTION
**Pull request by thijscobben**
This should migrate the records from V1 b2share.eudat.eu to any environment.

### CODE REVIEW

File: demo/b2share_demo/data/communities/euon.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
                             "type": "array",
                             "items": {
                                 "enum": [
+									"english",
```
Comment (emanueldima) [inline demo/b2share_demo/data/communities/euon.json]: One of this thing is not like the others 😄 . I'd map English to "Other"

File: demo/b2share_demo/data/communities/euon.json [hunk 2]
```diff
@@ -81,7 +85,8 @@
                                     "OntoStudio",
                                     "KAON",
                                     "PoolParty Thesaurus Server",
-                                    "Other..."
+                                    "Other...",
+									""
```
Comment (emanueldima) [inline demo/b2share_demo/data/communities/euon.json]: Empty entry is not needed if we have the "other"

File: demo/b2share_demo/cli.py [hunk 3]
```diff
@@ -106,19 +108,21 @@ def import_v1_data(verbose, download, token,
...
     if limit and not download:
         raise click.ClickException("Limit can only be set with download")
     if download:
-        filelist = os.listdir('.')
-        if len(filelist)>0:
-            raise click.ClickException("""You set download_dir to %s .
-            If you want to download files, download_dir should be an empty
-             directory.\n Please empty directory and try again.""" %
-             download_directory)
+        #filelist = os.listdir('.')
```
Comment (emanueldima) [inline demo/b2share_demo/cli.py]: should be uncommented?

File: demo/b2share_demo/helpers.py [hunk 4]
```diff
@@ -328,12 +344,20 @@ def download_v1_record(directory, record, verbose=False):
                 click.secho('    Download file "{}"'.format(file_dict.get('name')))
             filepath = os.path.join(directory, 'file_{}'.format(index))
             _save_file(file_dict['url'], filepath)
-            if int(os.path.getsize(filepath)) != int(file_dict.get('size')):
-                raise Exception("downloaded file size differs, {}".format(filepath))
+            #if int(os.path.getsize(filepath)) != int(file_dict.get('size')):
```
Comment (emanueldima) [inline demo/b2share_demo/helpers.py]: But we must make sure that the sizes are the same! It must be exactly the same data.

File: demo/b2share_demo/helpers.py [hunk 5]
```diff
@@ -408,24 +451,62 @@ def _process_record(rec):
...
-            finished = True
+            chunk = u.read(CHUNK_SIZE)
+            if chunk:
+                f.write(chunk)
+            else:
+                finished = True
         except:
-            current_app.logger.debug("IncompleteRead? %d" % counter)
-            counter += counter
+            finished = True
```
Comment (emanueldima) [inline demo/b2share_demo/helpers.py]: incorrect to assume that the file is correctly finished downloading, I'd simply quit with error message

File: demo/b2share_demo/data/communities/euon.json (hunk 2 above)
Comment (thijscobben) [inline demo/b2share_demo/data/communities/euon.json]: This is required to pass validation on usedOntologyEngineeringTool="" which appears and to my opinion we cannot map the case "No tool used" to "Other tool used" ? So I keep the value. Note that usedOntologyEngineeringTool is not a required field in this schema, so we can assume that "empty" indeed means "No tool used" (as would leaving out the key/value all together imply too).

File: demo/b2share_demo/data/communities/euon.json (hunk 1 above)
Comment (thijscobben) [inline demo/b2share_demo/data/communities/euon.json]: "english" was a value specified by one of the communities I advice to retain it in the schema this is their choice.

File: demo/b2share_demo/helpers.py (hunk 4 above)
Comment (thijscobben) [inline demo/b2share_demo/helpers.py]: fixed. This also voids the case you mention below:

File: demo/b2share_demo/helpers.py (hunk 5 above)
Comment (thijscobben) [inline demo/b2share_demo/helpers.py]: Fixed by uncommenting file size check.

**Pull request merged**